core/unit_test/TestGraph.hpp

```diff
@@ -539,6 +539,32 @@ TEST_F(TEST_CATEGORY_FIXTURE(graph), empty_graph_default_host_exec) {
   graph.get_execution_space().fence();
 }
 
+// Ensure that the graph always stores the node.
+// This is done by using a reference counted view as functor, and
+// checking that it is freed only upon graph destruction.
+TEST_F(TEST_CATEGORY_FIXTURE(graph), node_lifetime) {
+  const std::string alloc_label = "witness view for node lifetime";
+
+  std::optional<Kokkos::Experimental::Graph<TEST_EXECSPACE>> graph =
+      Kokkos::Experimental::create_graph(ex, [&](const auto& root) {
+        using view_t    = Kokkos::View<int[1], TEST_EXECSPACE>;
+        const auto node = root.then_parallel_for(
+            1, view_t(Kokkos::view_alloc(alloc_label, ex)));
```

Comment:
Holly smoke, this compiles?!

Reply:
Indeed. `Kokkos` does not check that the call operator we pass in returns `void`. Hacky hack that works, but as agreed I've made it less hacky.

Reply:
@dalg24 Do I simply close this thread because I'm not doing this anymore or do you want some additional action to be taken ?